packages/mwp-app-render/src/components/connectedWithMatchMedia.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { createStructuredSelector, createSelector } from 'reselect';
+import withMatchMedia from 'meetup-web-components/lib/utils/components/withMatchMedia';
+
+/**
+ * getConfig
+ * @param  {Object} state
+ * @return {Object} the config from state
+ */
+const getConfig = state => state.config || {};
+
+/**
+ * Checks if object is empty
+ * @param  {Object} object
+ * @return {Boolean} whether object is empty
+ */
+const isEmpty = object => object === undefined || Object.keys(object).length === 0;
+
+/**
+ * @param {Object} device - object containing device information
+ * @returns {Object} - default media object for `withMatchMedia`
+ */
+const getMediaOrDefault = (media, device) => {
+	if (!isEmpty(media)) {
+		return media;
+	}
+	if ( !isEmpty(device) ) {
+		const {isMobile, isTablet, isDesktop} = device;
+		const isTabletOrDesktop = isTablet || isDesktop;
+		return {
+			isAtSmallUp: Boolean(isMobile || isTabletOrDesktop),
+			isAtMediumUp: Boolean(isTabletOrDesktop),
+			isAtLargeUp: Boolean(isDesktop),
+		}
+	}
+	return {};
+};
+
+const mapStateToProps = createStructuredSelector({
+	device: createSelector(getConfig, config => (config.device || {})),
+});
+
+/**
+ * connectedWithMatchMediaComponent
+ * HOC that provides a default based on `device` in state.config
+ * if withMatchMedia has not yet detected the media
+ *
+ * `device` prop is provided from redux state in config.
+ * `media` prop is provided to the wrapped component in render().
+ *
+ * @returns {React.element}
+ */
+const connectedWithMatchMedia = WrappedComponent => {
+
+	/**
+	 * @module ConnectedWithMatchMedia
+	 */
+
+	class ConnectedWithMatchMedia extends React.Component {
+
+		render() {
+			const {device, media: currentMedia } = this.props;
+			const media = getMediaOrDefault(currentMedia, device);
+			return <WrappedComponent {...this.props} media={media}/>;
+		}
+	};
+	const wrappedComponentName =
+		WrappedComponent.displayName || WrappedComponent.name || 'Component';
+
+	ConnectedWithMatchMedia.WrappedComponent = WrappedComponent;
+	ConnectedWithMatchMedia.displayName = `ConnectedWithMatchMedia(${wrappedComponentName})`;
+	
+	/* connect to redux state */
+	const ConnectedComponent = connect(mapStateToProps)(ConnectedWithMatchMedia);
+	/* return connected wrapped component withMatchMedia */
+	return withMatchMedia(ConnectedComponent);
```

Comment:
withMatchMedia wraps the result of connecting the wrapped component. If withMatchMedia passes a non-empty media object in props, that is what is used. Otherwise we use a default media object based on device type.

Not sure if it matters whether withMatchMedia wraps the connected + wrapped component, which is what's happening here, or if we should wrap the wrapped component withMatchMedia and then wrap that with the connected component. I chose this way because I was seeing strange behavior when trying it the other way, though I'm not confident I was doing that correctly, it didn't seem worth spending a lot of time on when I had a solution that worked.

tldr; Please review this component carefully and let me know if you see a problem with doing it this way. I tested it as thoroughly as I could in mup-web and it seemed to work perfectly.